test/bigtest/tests/units/settings/MappingProfilesForm/TransformationsModal-test.js

```diff
@@ -148,30 +152,136 @@ describe('MappingProfilesTransformationsModal', () => {
+        });
+
+        describe('clicking on select all checkboxes', () => {
+          beforeEach(async () => {
+            await modal.transformations.selectAllCheckbox.clickInput();
+          });
+
+          it('should select all transformations (apply selection to unselected items)', () => {
+            expect(modal.transformations.checkboxes(0).isChecked).to.be.true;
+            expect(modal.transformations.checkboxes(1).isChecked).to.be.true;
+            expect(modal.transformations.checkboxes(2).isChecked).to.be.true;
+          });
+
+          it('should check select all checkboxes', () => {
+            expect(modal.transformations.selectAllCheckbox.isChecked).to.be.true;
+          });
+
+          it('should update total selected count', () => {
+            expect(modal.totalSelected.text).to.equal(totalSelectedMessage(3));
+          });
+        });
       });
 
-      it('should filter out the transformation list', () => {
-        expect(modal.transformations.list.rows(0).cells(1).text).to.equal(initialValues.transformations[1].displayName);
+      describe('clicking on select all checkboxes', () => {
+        beforeEach(async () => {
+          await modal.transformations.selectAllCheckbox.clickInput();
+        });
+
+        it('should deselect all transformations', () => {
+          expect(modal.transformations.checkboxes(0).isChecked).to.be.false;
+          expect(modal.transformations.checkboxes(1).isChecked).to.be.false;
+          expect(modal.transformations.checkboxes(2).isChecked).to.be.false;
+        });
+
+        it('should uncheck select all checkboxes', () => {
+          expect(modal.transformations.selectAllCheckbox.isChecked).to.be.false;
+        });
+
+        it('should update total selected count', () => {
+          expect(modal.totalSelected.text).to.equal(totalSelectedMessage(0));
+        });
       });
 
-      describe('turning on filters again', () => {
+      describe('turning off certain record type filters', () => {
         beforeEach(async () => {
           await modal.searchForm.recordTypeFilters(0).clickAndBlur();
           await modal.searchForm.recordTypeFilters(1).clickAndBlur();
         });
 
-        it('should display all transformation fields', () => {
-          expect(modal.transformations.list.rowCount).to.equal(initialValues.transformations.length);
-          expect(modal.resultsPane.header.sub).to.equal(`${initialValues.transformations.length} fields found`);
+        it('should display correct amount of transformation fields', () => {
+          expect(modal.transformations.list.rowCount).to.equal(1);
+          expect(modal.resultsPane.header.sub).to.equal('1 field found');
+        });
+
+        it('should filter out the transformation list', () => {
+          expect(modal.transformations.list.rows(0).cells(1).text).to.equal(initialValues.transformations[1].displayName);
+        });
+
+        describe('clicking on select all checkboxes', () => {
```

Comment:
Could you please add test case when
-- select all checkboxes is selected
-- then one item is unselected (so, in that case, all selected becomes unselected)
-- and then the filters are applied so that only that one item is hidden (so, in that case, all selected becomes selected again).

Basically it would be great to have a test for the last case when all selected becomes selected again.

Reply:
> Basically it would be great to have a test for the last case when all selected becomes selected again.

I'm not sure that I got your point here. Please, could you clarify it?

Reply:
Discussed.